### PR TITLE
Refactor signup flow

### PR DESCRIPTION
**Pull request by dbaq**
In this PR, I refactored the signup flow by:

- signup page has it is own dedicated controller
- use ```$urlRouterProvider.otherwise``` function to redirect to signup page if the user is not logged in or to the buyer page
- do "some" style on the signup page (needs to be finished with a better logo and background image)
- remove the MainCtrl and the main html page
- add a bunch of mixins (_mixins.scss) that will be useful for all the scss files

My changes impact views and controllers only, I haven't changed anything to the facebook login and to the login service, I am using the same functions.

Finally, this code is using the style guide (the one I talked about in #8).

Let me know what you think.

screen shot 2015-06-07 at 10 26 28 am

### CODE REVIEW

**Comment by valdrinkoshi** [comment]
Nice, I'll have a look asap and try locally before merging

**Comment by dbaq** [comment]
Since it is based on the style guide, you will have to run ```grunt serve --force``` otherwise the jshint task won't be happy.

**Comment by dbaq** [comment]
so wait for #9 before merging this PR

**Comment by valdrinkoshi** [comment]
Image works fine only for phone form factor, not for desktop/tablet. 
screen shot 2015-06-07 at 3 34 00 pm

File: app/scripts/signup.controller.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        vm.signIn = signIn;
+
+  
+        function signIn(loginType, user, pass) {
+            $ionicLoading.show({
+                template: 'Sign in...'
+            });
+            LoginService.login(loginType, user, pass).then(function didLogin() {
+                $ionicLoading.hide();
+                $state.go('app.buyer');
```
Comment (valdrinkoshi) [inline app/scripts/signup.controller.js]: Shouldn't state.go be called after ionicHistory.nextViewOptions?

File: app/templates/main.html [hunk 2]
```diff
@@ -1,65 +0,0 @@
...
-    </div>
-
-    <div ng-hide="ctrl.isLoggedIn">
-      <div class="item" ng-hide="(ctrl.doingLogin || ctrl.doingSignup)">
-        <!--<button class="button button-block button-positive" ng-click="ctrl.doingLogin=true">Log in</button>-->
-        <button class="button button-block button-positive" ng-click="ctrl.login('facebook')">Log in with Facebook
-        </button>
-        <!--<button class="button button-block button-positive" ng-click="ctrl.doingSignup=true">Sign up</button>-->
-      </div>
-      <form novalidate class="list list-inset" name="loginForm" ng-show="ctrl.doingLogin">
```
Comment (valdrinkoshi) [inline app/templates/main.html]: this was not used hence we can delete it, but we'll eventually need this when we'll re-enable the login/signup through home cooked. Reminder for myself

**Comment by valdrinkoshi** [comment]
All looks good :+1: 
Will merge this, adjustments can be done in another PR (e.g. image & button)

File: app/templates/main.html (hunk 2 above)
Comment (dbaq) [inline app/templates/main.html]: yep I confirmed with @marcandreoli yesterday if login with email was for the alpha, he said no, so I removed this code. I also have a css style for the buttons ```Sign up with email``` that we can reuse. Looks like the facebook one but grey.

**Comment by dbaq** [comment]
> Image & button works fine only for phone form factor, not for desktop/tablet. 

I am quite confused by what you're saying. I agree that's not responsive for tablet (it is just a simple media query to use) but I've seen with @marcandreoli that the tablet support it is not required for the alpha.

But about desktop, here you are showing a screenshot for desktop, and on the other PR you're talking about the browser back button.
The app is designed to be a mobile app, even if right now we are using chrome to develop the app. A lot of code will be specific for mobile purposes (pull-to-refresh, action sheet, etc0 and for cordova(phone camera, phone contacts). This thread on the forum explains my point http://forum.ionicframework.com/t/why-no-desktop/7908/6 (see the answer of mhartington).

So just to be clear (since @marcandreoli is reading Github notifications):
- the code on this repository should be only for a cordova webview for phones and tablets
- if we want a desktop version app, we will have to make a regular angularjs app or any other technology (we will be able to reuse some services and some code). We will have to use another CSS framework.

File: app/scripts/signup.controller.js (hunk 1 above)
Comment (dbaq) [inline app/scripts/signup.controller.js]: I haven't found anything about it, but for the readability, I agree that it should be after.

**Comment by valdrinkoshi** [comment]
Well no, the code should work in a generic browser (phone or not). I don't remember discussing about limiting the support to phone only (which phone btw? iphone only? android only?). Also, we won't limit the support of hybrid apps only, otherwise we'll be trapped in the circle of publishing a new app (app store, play store).
About mobile specific features, currently we don't have any. Action sheet behaves quite decently also on desktop. I don't think it will harm if we allow users to access homecooked also from their browser on desktop. We should support phone and tablet, and consider "tablet" experience also from desktop.
Anyways, i think this is going off-topic. I will merge this and we can do another PR for next adjustments.

**Comment by valdrinkoshi** [comment]
Thanks for the fixes @dbaq, i will merge these 2 PR and do another to fix the package.json i mentioned in the other PR ;)

**Comment by marcandreoli** [comment]
@dbaq  I will send you a new background image for the sign-in. Are there any specifications I should follow?
Thanks

**Comment by dbaq** [comment]
Yes and a new logo please, something that is retina friendly. I will give you multiple dimensions soon.

**Pull request merged**